### ISSUE TITLE
linter: `array-callback-return` false-positive

### ISSUE THREAD

**Issue by ai** [oxc-project/oxc#21485]
### What version of Oxlint are you using?

1.60.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "rules": {
    "array-callback-return": "error"
  }
}
```

### What happened?

```ts
function filterNested(allItems, group) {
  return Array.from(allItems).filter(item => {
    let current = item.parentElement
    while (current && current !== group) {
      if (current.getAttribute('aria-hidden') === 'true') {
        return false
      }
      current = current.parentElement
    }
    return true
  })
}
```

generates error:

```
eslint(array-callback-return): Callback for array method "Array.prototype.filter" does not return on all code paths
```

But all paths has `return` and [ESLint does not report the same issue](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IGFycmF5LWNhbGxiYWNrLXJldHVybjogXCJlcnJvclwiICovXG5mdW5jdGlvbiBmaWx0ZXJOZXN0ZWQoYWxsSXRlbXMsIGdyb3VwKSB7XG4gIHJldHVybiBBcnJheS5mcm9tKGFsbEl0ZW1zKS5maWx0ZXIoaXRlbSA9PiB7XG4gICAgbGV0IGN1cnJlbnQgPSBpdGVtLnBhcmVudEVsZW1lbnRcbiAgICB3aGlsZSAoY3VycmVudCAmJiBjdXJyZW50ICE9PSBncm91cCkge1xuICAgICAgY3VycmVudCA9IGN1cnJlbnQucGFyZW50RWxlbWVudFxuICAgIH1cbiAgICByZXR1cm4gdHJ1ZVxuICB9KVxufSIsIm9wdGlvbnMiOnsicnVsZXMiOnsiYXJyYXktY2FsbGJhY2stcmV0dXJuIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

[Reproduction](https://playground.oxc.rs/?lintRules=eslint%2Farray-callback-return&options={"run"%3A{"lint"%3Atrue%2C"formatter"%3Afalse%2C"transform"%3Afalse%2C"isolatedDeclarations"%3Afalse%2C"whitespace"%3Afalse%2C"mangle"%3Afalse%2C"compress"%3Afalse%2C"scope"%3Atrue%2C"symbol"%3Atrue%2C"cfg"%3Afalse}%2C"parser"%3A{"extension"%3A"tsx"%2C"allowReturnOutsideFunction"%3Atrue%2C"preserveParens"%3Atrue%2C"allowV8Intrinsics"%3Atrue%2C"semanticErrors"%3Atrue}%2C"linter"%3A{}%2C"formatter"%3A{"useTabs"%3Afalse%2C"tabWidth"%3A2%2C"endOfLine"%3A"lf"%2C"printWidth"%3A80%2C"singleQuote"%3Afalse%2C"jsxSingleQuote"%3Afalse%2C"quoteProps"%3A"as-needed"%2C"trailingComma"%3A"all"%2C"semi"%3Atrue%2C"arrowParens"%3A"always"%2C"bracketSpacing"%3Atrue%2C"bracketSameLine"%3Afalse%2C"objectWrap"%3A"preserve"%2C"singleAttributePerLine"%3Afalse}%2C"transformer"%3A{"target"%3A"es2015"%2C"useDefineForClassFields"%3Atrue%2C"experimentalDecorators"%3Atrue%2C"emitDecoratorMetadata"%3Atrue}%2C"isolatedDeclarations"%3A{"stripInternal"%3Afalse}%2C"codegen"%3A{"normal"%3Atrue%2C"jsdoc"%3Atrue%2C"annotation"%3Atrue%2C"legal"%3Atrue}%2C"compress"%3A{}%2C"mangle"%3A{"topLevel"%3Atrue%2C"keepNames"%3Afalse}%2C"controlFlow"%3A{"verbose"%3Afalse}%2C"inject"%3A{"inject"%3A{}}%2C"define"%3A{"define"%3A{}}}&code=function+filterNested(allItems%2C+group)+{%0A++return+Array.from(allItems).filter(item+%3D>+{%0A++++let+current+%3D+item.parentElement%0A++++while+(current+%26%26+current+!%3D%3D+group)+{%0A++++++current+%3D+current.parentElement%0A++++}%0A++++return+true%0A++})%0A})
